src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                raise Exception(
+                    "The dell-environment.yaml file does not exist")
+            cmds = ['sed -i "s|  # NovaSchedulerDefaultFilters|  ' +
+                    'NovaSchedulerDefaultFilters|" ' + file_path]
+            cmds.append(
+                'sed -i "s|dellnfv::hugepages::enable:.*'
+                '|dellnfv::hugepages::enable: ' +
+                str(enable_hugepage) +
+                '|" ' +
+                file_path)
+            cmds.append(
+                'sed -i "s|dellnfv::numa::enable:.*|dellnfv::numa::enable: ' +
+                str(enable_numa) +
+                '|" ' +
+                file_path)
+
+            cmds.append(
+                'sed -i "s|DellComputeCount:.*|DellComputeCount: ' +
+                str(dell_compute_count) +
+                '|" ' +
+                file_path)
+
+            if enable_hugepage:
+                hugepage_number = ConfigOvercloud.calculate_hugepage_count(
+                    hugepage_size)
+                cmds.append('sed -i "s|dellnfv::hugepages::hugepagesize:.*'
+                            '|dellnfv::hugepages::hugepagesize: ' +
+                            hugepage_size[0:-1] + '|" ' + file_path)
+                cmds.append(
+                    'sed -i "s|dellnfv::hugepages::hugepagecount:.*'
+                    '|dellnfv::hugepages::hugepagecount: ' +
+                    str(hugepage_number) +
+                    '|" ' +
+                    file_path)
+            if enable_numa:
+                vcpu_pin_set = ConfigOvercloud.calculate_hostos_cpus(
+                    hostos_cpu_count)
+                cmds.append(
+                    "sed -i 's|dellnfv::numa::vcpu_pin_set:.*"
+                    "|dellnfv::numa::vcpu_pin_set: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+                cmds.append(
+                    'sed -i "s|  # NovaVcpuPinSet|  ' +
+                    'NovaVcpuPinSet|" ' +
+                    file_path)
+                cmds.append(
+                    "sed -i 's|NovaVcpuPinSet:.*|NovaVcpuPinSet: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+
+            for cmd in cmds:
+                status = os.system(cmd)
+                logger.debug("cmd: {}".format(cmd))
+            if status != 0:
+                raise Exception(
+                    "Failed to execute the command {}"
+                    " with error code {}".format(
+                        cmd, status))
+
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to modify the dell_environment.yaml"
+                " at location {}".format(file_path))
+
+    def get_dell_compute_nodes_hostnames(self, nova):
```

Comment:
I can't find any place where this method is called.  Is it needed?

Reply:
Fixed